system/mp/mpcomm.cpp

```diff
@@ -716,7 +728,14 @@ protected: friend class CMPPacketReader;
                 hostep.setLocalHost(parent->getPort());
                 id[0].set(hostep);
                 id[1].set(remoteep);
+
+                unsigned long addrval = DIGIT1*id[0].ip[0] + DIGIT2*id[0].ip[1] + DIGIT3*id[0].ip[2] + DIGIT4*id[0].ip[3] + id[0].port;
```

Comment:
This overflows a long .. would need to be a int64 if doing this way..

But I think it would be better if you added a compare operator to SocketEndpointV4 and pass that to attachSocket